packages/compass-schema/src/modules/schema-analysis.spec.js

```diff
@@ -3,6 +3,32 @@ import bson from 'bson';
 import { expect } from 'chai';
 import { analyzeSchema } from './schema-analysis';
 
+function schemaToPaths(fields, parent = []) {
+  const paths = [];
+
+  for (const field of fields) {
+    const path = [...parent, field.name];
+    paths.push(path);
+
+    // recurse on doc
+    const doc = field.types.find((f) => f.bsonType === 'Document');
+    if (doc) {
+      paths.push(...schemaToPaths(doc.fields, path));
+    }
+
+    // recurse on array
+    const array = field.types.find((f) => f.bsonType === 'Array');
+    if (array) {
+      const arrayDoc = array.types.find((f) => f.bsonType === 'Document');
+      if (arrayDoc) {
+        paths.push(...schemaToPaths(arrayDoc.fields, [...path, '*']));
```

Comment:
I suppose the `*` isn't technically necessary and we can just know that path components inside arrays mean object fields later when we build the json doc or csv row. Undecided about it.